### DISCUSSION TITLE
UnicodeDecodeError 'utf-8' codec can't decode byte 0xc3

### DISCUSSION THREAD

**Discussion by bburbidge** [oracle/python-oracledb#33] [Q&A] [answered]
I have the code as follows:

    # create Session pool
    pool = oracledb.create_pool(user=data['oracle_user'], password=data['oracle_pw'], dsn=data['oracle_server_ip'], min=2, max=5, increment=1)
    # Create SQLAlchemy engine as preferred by Pandas & Python
    engine = sqlalchemy.create_engine("oracle://", creator=pool.acquire, poolclass=sqlalchemy.pool.NullPool)

    with engine.connect() as conn:
         df:pandas.DataFrame = pandas.read_sql(query_string, conn)

However, when running one particular query, I'm getting a "UnicodeDecodeError: 'utf-8' codec can't decode byte" exception. I see from the documentation page that encoding has defaulted to UTF-8. Any ideas on how to resolve this exception programmatically without making changes to the database?

**Comment by anthony-tuininga** [comment, marked as answer]
Take a look at this [documentation](https://python-oracledb.readthedocs.io/en/latest/user_guide/sql_execution.html#querying-corrupt-data). You should ignore the comment about using UTF-8 as that is stale from the conversion from cx_Oracle. python-oracledb uses UTF-8 internally at all times now!

**Comment by cjbj** [comment]
Also see the documentation [Fetching Raw Data](https://python-oracledb.readthedocs.io/en/latest/user_guide/sql_execution.html#fetching-raw-data).
